ncat/ncat_connect.c

```diff
@@ -863,43 +847,40 @@ static int do_proxy_socks5(void)
             break;
         case 1:
             loguser("Error: general SOCKS server failure.\n");
-            close(sd);
-            return -1;
+            goto error;
         case 2:
             loguser("Error: connection not allowed by ruleset.\n");
-            close(sd);
-            return -1;
+            goto error;
         case 3:
             loguser("Error: Network unreachable.\n");
-            close(sd);
-            return -1;
+            goto error;
         case 4:
             loguser("Error: Host unreachable.\n");
-            close(sd);
-            return -1;
+            goto error;
         case 5:
             loguser("Error: Connection refused.\n");
-            close(sd);
-            return -1;
+            goto error;
         case 6:
             loguser("Error: TTL expired.\n");
-            close(sd);
-            return -1;
+            goto error;
         case 7:
             loguser("Error: Command not supported.\n");
-            close(sd);
-            return -1;
+            goto error;
         case 8:
             loguser("Error: Address type not supported.\n");
-            close(sd);
-            return -1;
+            goto error;
         default:
             loguser("Error: unassigned value in the reply.\n");
-            close(sd);
-            return -1;
+            goto error;
     }
 
     return(sd);
+error:
+    if (proxy_auth != NULL)
+        free(proxy_auth);
```

Comment:
I think you should ```free(proxy_auth)``` also when no error occurs.  This is not a problem introduced by your patch but still worth fixing.